src/MSBuild/XMake.cs

```diff
@@ -1427,17 +1532,37 @@ private static void ResetGatheringSwitchesState()
                             {
                                 if (graphBuildOptions != null)
                                 {
-                                    (result, exception) = ExecuteGraphBuild(buildManager, graphBuildRequest);
+                                    graphResult = ExecuteGraphBuild(buildManager, graphBuildRequest);
+
+                                    if (saveProjectResult)
+                                    {
+                                        ProjectGraphEntryPoint entryPoint = graphBuildRequest.ProjectGraphEntryPoints.Single();
+                                        if (!entryPoint.GlobalProperties.ContainsKey(PropertyNames.IsGraphBuild))
+                                        {
+                                            entryPoint.GlobalProperties[PropertyNames.IsGraphBuild] = "true";
+                                        }
+
+                                        result = graphResult.ResultsByNode.Single(
+                                            nodeResultKvp =>
+                                            nodeResultKvp.Key.ProjectInstance.FullPath.Equals(entryPoint.ProjectFile) &&
+                                            nodeResultKvp.Key.ProjectInstance.GlobalProperties.Count == entryPoint.GlobalProperties.Count &&
+                                            nodeResultKvp.Key.ProjectInstance.GlobalProperties.All(propertyKvp => entryPoint.GlobalProperties[propertyKvp.Key].Equals(propertyKvp.Value)))
```

Comment:
The dictionary indexer throws `KeyNotFoundException` if the key is not found. Is it ok to assume that `nodeResultKvp.Key.ProjectInstance.GlobalProperties` and `entryPoint.GlobalProperties` have exactly the same keys here?

Reply:
It was not. I switched to `TryGetValue`.

before:

```
MSBUILD : error MSB1025: An internal failure occurred while running MSBuild.
This is an unhandled exception in MSBuild -- PLEASE OPEN A BUG AGAINST THE MSBUILD TEAM.
System.Collections.Generic.KeyNotFoundException: The given key 'TargetFramework' was not present in the dictionary.
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at Microsoft.Build.CommandLine.MSBuildApp.<>c__DisplayClass29_0.<BuildProject>b__1(KeyValuePair`2 propertyKvp) in S:\msbuild\src\MSBuild\XMake.cs:line 1548
   at System.Linq.Enumerable.All[TSource](IEnumerable`1 source, Func`2 predicate)
   at Microsoft.Build.CommandLine.MSBuildApp.<>c__DisplayClass29_0.<BuildProject>b__0(KeyValuePair`2 nodeResultKvp) in S:\msbuild\src\MSBuild\XMake.cs:line 1547
   at System.Linq.Enumerable.TryGetSingle[TSource](IEnumerable`1 source, Func`2 predicate, Boolean& found)
   at System.Linq.Enumerable.Single[TSource](IEnumerable`1 source, Func`2 predicate)
   at Microsoft.Build.CommandLine.MSBuildApp.BuildProject(String projectFile, String[] targets, String toolsVersion, Dictionary`2 globalProperties, Dictionary`2 restoreProperties, ILogger[] loggers, LoggerVerbosity verbosity, DistributedLoggerRecord[] distributedLoggerRecords, Int32 cpuCount, Boolean enableNodeReuse, TextWriter preprocessWriter, TextWriter targetsWriter, Boolean detailedSummary, ISet`1 warningsAsErrors, ISet`1 warningsNotAsErrors, ISet`1 warningsAsMessages, Boolean enableRestore, ProfilerLogger profilerLogger, Boolean enableProfiler, Boolean interactive, ProjectIsolationMode isolateProjects, GraphBuildOptions graphBuildOptions, Boolean lowPriority, Boolean question, String[] inputResultsCaches, String outputResultsCache, Boolean saveProjectResult, BuildResult& result, String[] commandLine) in S:\msbuild\src\MSBuild\XMake.cs:line 1545
```

now:

```sh-session
❯ .\.dotnet\dotnet.exe msbuild -bl -graph S:\play\mt_graph\mt_graph.csproj -getTargetResult:Build
{
  "TargetResults": {
    "Build": {
      "Result": "Success",
      "Items": []
    }
  }
}
```